### PR TITLE
drivers/ads101x: update I2C API

### PR DESCRIPTION
**Pull request by ZetaR60**
This is a WIP update to ads101x's I2C API usage. Currently untested due to lack of API update to atmega_common. See also: #6577 and #9162

### CODE REVIEW

**Comment by ZetaR60** [comment]
I may need to rebase this on new_i2c_if

**Comment by dylad** [comment]
Hi @ZetaR60 
Thanks for your PR. Unfortunately this PR cannot go to the master branch. It belongs to the new_i2c_if branch.
Once all the job will be done in new_i2c_if branch, the whole branch will be merge to master.
We're doing this way to prevent the CI from complaining.

**Comment by ZetaR60** [comment]
@dylad I just had to rebase this on new_i2c_if, because switching it over brought a bunch of stuff from master with it. Should be good now.

**Comment by ZetaR60** [comment]
Should be up to date again.

File: drivers/ads101x/ads101x.c [hunk 1]
```diff
@@ -67,24 +63,35 @@ int ads101x_alert_init(ads101x_alert_t *dev,
...
+static int _ads101x_init_test(i2c_t i2c, uint8_t addr)
 {
     uint8_t regs[2];
 
-    /* Acquire test */
     i2c_acquire(i2c);
-    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+
+    /* Register read test */
+    if (i2c_read_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2, I2C_NOSTOP)
```
Comment (dylad) [inline drivers/ads101x/ads101x.c]: I don't see any reason here to have a NOSTOP flag. (same at other place in the code)
It's not forbidden to read some regs send a STOP and then send data into another register.

File: drivers/ads101x/ads101x.c [hunk 2]
```diff
@@ -127,22 +134,21 @@ int ads101x_read_raw(const ads101x_t *dev, int16_t *raw)
     i2c_acquire(I2C);
 
     /* Read control register */
-    i2c_read_regs(I2C, ADDR, ADS101X_CONF_ADDR, &regs, 2);
```
Comment (dylad) [inline drivers/ads101x/ads101x.c]: NOSTOP required ?

File: drivers/ads101x/ads101x.c [hunk 3]
```diff
@@ -103,7 +110,7 @@ int ads101x_set_mux_gain(const ads101x_t *dev, uint8_t mux_gain)
 
     i2c_acquire(I2C);
 
-    i2c_read_regs(I2C, ADDR, ADS101X_CONF_ADDR, &regs, 2);
```
Comment (dylad) [inline drivers/ads101x/ads101x.c]: NOSTOP required ?

File: drivers/ads101x/ads101x.c [hunk 4]
```diff
@@ -67,24 +63,35 @@ int ads101x_alert_init(ads101x_alert_t *dev,
...
     }
 
-    /* Register read/write test */
-    i2c_read_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2);
     regs[1] = (regs[1] & ~ADS101X_DATAR_MASK) | ADS101X_DATAR_3300;
 
-    i2c_write_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2);
-    i2c_read_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2);
+    /* Register write test */
+    if (i2c_write_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2, I2C_NOSTOP)
```
Comment (dylad) [inline drivers/ads101x/ads101x.c]: same here, is the NOSTOP really useful ?

**Comment by ZetaR60** [comment]
NOSTOPs removed.

**Comment by mali** [comment]
@ZetaR60 , I'll push an atmega_common api update (WIP) soon. So you'll be able to test.

**Comment by ZetaR60** [comment]
@mali I am not seeing the changes to periph/i2c.c: https://github.com/mali/RIOT/blob/atmega_new_i2c_if/cpu/atmega_common/periph/i2c.c

**Comment by mali** [comment]
Oups .. sorry, it seems I've done something wrong :-/

**Comment by mali** [comment]
It's repaired ! see also #9252

**Comment by ZetaR60** [comment]
I tried tests/driver_ads101x with #9252 and got this error (debug enabled for ads101x and gpio.c):

```
[liveuser@localhost-live driver_ads101x]$ make BOARD=mega-xplained term
/home/liveuser/Desktop/RIOT_ads101x_i2c_api_merge/dist/tools/pyterm/pyterm -p "/dev/ttyUSB0" -b "9600"
No handlers could be found for logger "root"
2018-05-30 23:06:06,059 - INFO # Connect to serial port /dev/ttyUSB0
Welcome to pyterm!
Type '/exit' to exit.
2018-05-30 23:06:07,891 - INFO # 

2018-05-30 23:06:07,987 - INFO # main(): This is RIOT! (Version: 2018.07-devel-274-gf8bc-localhost-live-RIOT_ads101x_i2c_api)
2018-05-30 23:06:08,035 - INFO # ADS101X analog to digital driver test application
2018-05-30 23:06:08,035 - INFO # 
2018-05-30 23:06:08,131 - INFO # Initializing ADS101x analog to digital at I2C_DEV(0)... START condition transmitted
2018-05-30 23:06:08,147 - INFO # I2C Status is: START
2018-05-30 23:06:08,179 - INFO # ADDRESS and FLAG transmitted
2018-05-30 23:06:08,211 - INFO # NACK has been received for ADDRESS
2018-05-30 23:06:08,243 - INFO # STOP condition transmitted
2018-05-30 23:06:08,291 - INFO # [ads101x] init - error: unable to read reg 1
2018-05-30 23:06:08,291 - INFO # [Failed]
/exit
2018-05-30 23:06:11,546 - INFO # Exiting Pyterm
[liveuser@localhost-live driver_ads101x]$
```

I will continue investigation tomorrow.

**Comment by ZetaR60** [comment]
The specific NACK received is 0x20. I have not been able to track down the problem yet. The odd thing is that trunk has almost the same execution path as the new code, but trunk works fine and the new code is failing. The main difference is moving i2c_init, but moving this back to ads101x.c does not fix the problem.

File: drivers/ads101x/ads101x.c (hunk 2 above)
Comment (smlng) [reply drivers/ads101x/ads101x.c]: why should it be *required*?

File: drivers/ads101x/ads101x.c [hunk 5]
```diff
@@ -67,24 +65,35 @@ int ads101x_alert_init(ads101x_alert_t *dev,
...
+static int _ads101x_init_test(i2c_t i2c, uint8_t addr)
 {
     uint8_t regs[2];
 
-    /* Acquire test */
     i2c_acquire(i2c);
-    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+
+    /* Register read test */
+    if (i2c_read_regs(i2c, addr, ADS101X_CONF_ADDR, &regs, 2, 0x0)
```
Comment (smlng) [inline drivers/ads101x/ads101x.c]: I mostly saw a simple `0` for the flags in other places, IMHO we should stick to one style.

Comment (ZetaR60) [reply drivers/ads101x/ads101x.c]: I use 0 for defaults that are numerical, 0x0 for bit fields, and NULL for pointers. This is consistent with my prior work. IMO asking for developers to conform to a community standard for such detailed stylistic choices is burdensome without significant benefit.

Comment (smlng) [reply drivers/ads101x/ads101x.c]: agreed, maybe the (most) correct way would be to use the `I2C_FLAG_NONE` to explicitly state that no flag should be set or used here. But we could leave that to a cleanup PR over all drivers and stuff later on.

File: drivers/ads101x/ads101x_saul.c [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2017 OTA keys S.A.
- *               2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *               2018 Acutam Automation, LLC
```
Comment (smlng) [inline drivers/ads101x/ads101x_saul.c]: unrelated, and unclear why? Here and other files

**Comment by dylad** [comment]
@ZetaR60 atmega I2C driver has been merge, do you think you can re-test this PR ?

File: drivers/ads101x/ads101x_saul.c (hunk 6 above)
Comment (ZetaR60) [reply drivers/ads101x/ads101x_saul.c]: I have transferred my copyrights on RIOT code from myself (Matthew Blue) to my company (Acutam Automation, LLC). I figured it was a minor enough change that it did not require a separate PR, or detailed explanation.

**Comment by ZetaR60** [comment]
Rebased to pick up ATmega changes.

**Comment by ZetaR60** [comment]
Everything looks okay. Let me know when to squash.

```
user@user ~/Desktop/RIOT_ads101x_i2c_api/tests/driver_ads101x $ make BOARD=mega-xplained term
/home/user/Desktop/RIOT_ads101x_i2c_api/dist/tools/pyterm/pyterm -p "/dev/ttyUSB0" -b "9600"
2018-07-04 13:07:06,775 - INFO # Connect to serial port /dev/ttyUSB0
Welcome to pyterm!
Type '/exit' to exit.
2018-07-04 13:07:08,696 - INFO #  

2018-07-04 13:07:08,776 - INFO # main(): This is RIOT! (Version: 2018.07-devel-409-gd2631-user-RIOT_ads101x_i2c_api)
2018-07-04 13:07:08,840 - INFO # ADS101X analog to digital driver test application
2018-07-04 13:07:08,840 - INFO # 
2018-07-04 13:07:08,904 - INFO # Initializing ADS101x analog to digital at I2C_DEV(0)... [OK]
2018-07-04 13:07:08,952 - INFO # Initializing ADS101x alert at I2C_DEV(0)... [OK]
2018-07-04 13:07:08,984 - INFO # Enabling alert interrupt: [OK]
2018-07-04 13:07:09,064 - INFO # Raw analog read. CH0: 5456 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,192 - INFO # Raw analog read. CH0: 5584 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,320 - INFO # Raw analog read. CH0: 5488 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,432 - INFO # Raw analog read. CH0: 5536 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,560 - INFO # Raw analog read. CH0: 5616 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,672 - INFO # Raw analog read. CH0: 5536 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,800 - INFO # Raw analog read. CH0: 5472 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:09,928 - INFO # Raw analog read. CH0: 5440 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,040 - INFO # Raw analog read. CH0: 5616 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,167 - INFO # Raw analog read. CH0: 5520 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,279 - INFO # Raw analog read. CH0: 5520 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,407 - INFO # Raw analog read. CH0: 5536 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,535 - INFO # Raw analog read. CH0: 5584 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,647 - INFO # Raw analog read. CH0: 5632 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,775 - INFO # Raw analog read. CH0: 5488 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:10,887 - INFO # Raw analog read. CH0: 5600 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,034 - INFO # Raw analog read. CH0: 5520 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,143 - INFO # Raw analog read. CH0: 5568 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,255 - INFO # Raw analog read. CH0: 5488 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,383 - INFO # Raw analog read. CH0: 5504 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,495 - INFO # Raw analog read. CH0: 5616 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,623 - INFO # Raw analog read. CH0: 5488 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,751 - INFO # Raw analog read. CH0: 5632 CH1: 18800 CH2: 32752 CH3: 32752
2018-07-04 13:07:11,863 - INFO # Raw analog read. CH0: 5472 CH1: 18800 CH2: 32752 CH3: 32752
```

File: drivers/ads101x/ads101x_saul.c (hunk 6 above)
Comment (smlng) [reply drivers/ads101x/ads101x_saul.c]: ah okay, sorry was confused by GitHub username vs author name, never mind.

**Comment by dylad** [comment]
Here we go !

**Pull request merged**